src/import/codegen.ts

```diff
@@ -213,3 +213,130 @@ export function generateConstruct(typegen: TypeGenerator, def: ApiObjectDefiniti
     }
   });
 }
+
+/**
+ * Emit imports for generated helm construct
+ * @param code CodeMaker istance
+ */
+export function emitHelmHeader(code: CodeMaker) {
+  code.line('// generated by cdk8s');
+  code.line('import { Helm } from \'cdk8s\';');
+  code.line('import { Construct } from \'constructs\';');
+  code.line();
+}
+
+/**
+ * Helm Object Definition
+ */
+export interface HelmObjectDefinition {
+  /**
+   * `values.schema.json` for the helm chart
+   */
+  readonly schema: JSONSchema4 | undefined;
+  /**
+   * Chart name
+   */
+  readonly chartName: string;
+  /**
+   * Chart url
+   */
+  readonly chartUrl: string;
+  /**
+   * Chart version
+   */
+  readonly chartVersion: string;
+  /**
+   * Chart dependencies
+   */
+  readonly chartDependencies: string[];
+  /**
+   * Fully qualified name for the construct
+   */
+  readonly fqn?: string;
+}
+
+export function generateHelmConstruct(typegen: TypeGenerator, def: HelmObjectDefinition) {
+
+  const chartName = TypeGenerator.normalizeTypeName(def.chartName);
+  const schema = def.schema;
+  const repoUrl = def.chartUrl;
+  const chartVersion = def.chartVersion;
+
+  // Create custom type
+  typegen.emitCustomType(chartName, code => {
+
+    const valuesInterface = `${chartName}ValuesProps`;
+    if (schema !== undefined) {
+      // Interface for schema generated props
+      let schemaGenValuesInterface: string = 'SchemaGeneratedValues';
+      schemaGenValuesInterface = typegen.emitType(schemaGenValuesInterface, schema, def.fqn);
+
+      // Creating values interface
+      emitValuesInterface();
+
+      function emitValuesInterface() {
+        code.openBlock(`export interface ${valuesInterface} extends ${schemaGenValuesInterface}`);
+
+        // Sub charts or dependencies
+        for (const dependency of def.chartDependencies) {
+          code.line(`readonly ${dependency}?: { [key: string]: any };`);
+        }
+
+        // Global values
+        code.line('readonly global?: { [key: string]: any };');
+
+        code.closeBlock();
+      }
+    }
+
+    // Creating construct properties
+    emitPropsInterface();
+
+    code.line();
+
+    // Creating construct for helm chart
+    emitConstruct();
+
+    function emitPropsInterface() {
+      code.openBlock(`export interface ${chartName}Props`);
+
+      code.line('readonly namespace?: string;');
+      code.line('readonly releaseName?: string;');
+      code.line('readonly helmExecutable?: string;');
+      code.line('readonly helmFlags?: string[];');
+
+      if (schema === undefined) {
+        code.line('readonly values?: { [key: string]: any };');
```

Comment:
Would be really nice to check if there are any required properties in the schema, and then mark `values` as required - which would then need to be propagated to the entire `props` as required.